improver/utilities/save.py

```diff
@@ -158,6 +158,7 @@ def save_netcdf(
         )
 
     # save atomically by writing to a temporary file and then renaming
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
```

Comment:
Update doc-string to make it clear that this will happen.

Reply:
This looks like it will be hitting the metadata server an additional time for each call to save a file. Is that a concern? I imagine it's fine, but just checking.

Reply:
For current configurations (backwards compatibility) that's true (not thought about that).

For new configurations I'm not sure it will since it simply moves this directory creation from rose to here (unless it wasn't done per file??).  Well either way, new configurations don't save files from all steps, only those with explicit save steps so I don't think such things are a concern there.

The eventual goal would be to move save/load away from IMPROVER since it should be project/site owned.  To that end, right now, to aid backward compatibility, barebones demonstrates defining a Save proc. module (aka plugin) which simply calls IMPROVERS existing netcdf save.  Ultimately though, this entire function should be moved out.

Hmmm, I think I'll move this `os.mkdirs` to the barebones Save instead.  This way such a change wont effect existing configurations in any way.

Good shout both 👍 
A change I considered trivial here is clearly a significant enough behaviour change.